web/docs/deployment/database.md

```diff
@@ -0,0 +1,11 @@
+---
+title: Database
+---
+
+TODO: we'll explain the database in the context of deployment
+
+TODO: the only requirement from Wasp's point of view - it has to be a PostgreSQL database accessible from the server via the `DATABASE_URL` environment variable
+
+TODO: we'll talk about migrations: how are they created in development, how are they applied in production, how to debug failed migrations
+
+TODO: we'll talk about how to open the DB studio in development and production
```

Comment:
The main goal of this docs rewrite it to provide the users one big story about going into production from start to finish. The idea was: let's talk about all the things you need to understand so you know how to deploy it successfully. For that reason, I'm maybe repeating bit of content in the deployment docs (not always, for env vars docs I ink to the env docs from production env docs).

**Your second variant: A.P, B.P and then P links to them** is what we mostly had today and we got feedback from people that they didn't know where to look. I get it, if we link from P to them, they will find the same info, but then they have to jump around. 

So, what I'm saying, I'm more for _writing the best deployment docs_ we can now, deduplicate where it's obvious (like env vars) and optimise the structure once we are done with most of the docs.

Reply:
I get what you mean hm, but I don't like the duplication. It makes docs hard to maintain and keep in sync, and we will again end up with people having trouble following the docs, because information is repeated so it is never clear if you got all the info on the topic or not when reading the docs, instead you have to hunt it down through multiple parts of the docs, there might be discrepancies, ... . Sometimes some duplication is neccessary of course, but we should really aim to keep it to minimum.

Did we really get feedback that problem is that we link too much out of deployment part of the docs? Because my feeling was that the problem is not that, but not enough structure in the deployment docs. So far we didn't have great deployment docs that link outside, we had not so good deployment docs that link outside. My bet is if we organize them really well now, with nice structure and flow, that it won't matter if they often link to A.P and B.P. Why would it matter at the end, why would that be the problem? If structure is clear and reasonable, and it is clear which links I am expected to follow, why wouldn't that work? Or the opposite, A linking to P.A. I think it might depend on the feature what makes the most sense.

Ok yeah but I can agree with the approach you proposed -> we can write the best docs we can now, and the dedupe them. That does make sense to me, if it helps you to write the first version of deployment docs. That said, I wouldn't want us to say we will do deduping later, I would consider the whole thing not done till deduping is not done, I believe you also intended to do it like that? In that case, I would maybe advise leaving comments right now, as you are writing docs, on the parts that are likely or certainly duplication, so we have an easier way later hunting them down?

Reply:
I'll label stuff with comments:
```
<!-- TOPIC: env vars -->
```

So we can through all topic mentions and see if we there things mentioned already elsewhere or if it be good to move the thing to the non-deployment part of docs.

> Did we really get feedback that problem is that we link too much out of deployment part of the docs?
Yes, sir! I had link to three different pages to explain something on our Discord related to deployment. I can't remember what, but I know that one case. 

The reading flow gets interrupted if you have to go like this:
- _We'll talk about deployment ..._
- _Build the app like this ..._
- Now go to _link_ to read about env vars in production
- Now go to _link_ to read about database setup in production
- _Now let's do the next step ..._ 

I vote for people understanding and deploying Wasp apps vs. worrying about duplication too much. If some stuff is repeated verbatim - it's going into a snippet that we put in both places :)

Reply:
Maybe snippets are a good soltion, but I wouldn't underestimate duplication -> it can easily result in mistakes in docs + it will make it harder and more tedious for us to update them.

I am now nitpicking a bit, but what you wrote above about getting feedback taht probelms is that we link too much out of deployment part of the docs -> you didn't show that you got such feedback, instead you described a situation where you had to link to multiple different places while providing and answer. And sure, it breaks the flow, but it is big topic, and those are big chunks, so it might not be much of an issue really. Did anybody say that it is an issue, or give some other signs / feedback that made you think this is an issue, or is it your assumption that this is an issue?

Reply:
> Maybe snippets are a good soltion, but I wouldn't underestimate duplication -> it can easily result in mistakes in docs + it will make it harder and more tedious for us to update them.
> 
> I am now nitpicking a bit, but what you wrote above about getting feedback taht probelms is that we link too much out of deployment part of the docs -> you didn't show that you got such feedback, instead you described a situation where you had to link to multiple different places while providing and answer. And sure, it breaks the flow, but it is big topic, and those are big chunks, so it might not be much of an issue really. Did anybody say that it is an issue, or give some other signs / feedback that made you think this is an issue, or is it your assumption that this is an issue?

It was a conversation with user named Sven on our Discord where he expressed confusion with 3 different parts of the docs he had to read to understand one deployment concept I was trying to convey. I don't remember the specifics so I can't really find it that fast.

Reply:
Hm ok, although I am guessing those were env vars and they were indeed explained on multiple places, wiht overlaps, and then some stuff missing. Which is kind of exaclty what I am trying to avoid, I would rather have stuff explained in a central place and then linked to, then having multiple explanaitons that drift with time and it is not clear which one is exaustive.

You intuition does mean the most in this case, so all right, let's follow that.